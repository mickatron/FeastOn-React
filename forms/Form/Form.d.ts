import * as React from 'react';

export interface FormProps {
  [key: string]: any;
  /** An element type to render as (string or function). */
  as?: any;
  /** Primary content. */
  children?: React.ReactNode;
  /** Additional classes. */
  className?: string | array | object;
  /** BEM modifier attributes.*/
  modifier?: string | array;
  /** Width className */
  width?: string;
}

declare class Form extends React.Component<FormProps, {}> {
}

export default Form;