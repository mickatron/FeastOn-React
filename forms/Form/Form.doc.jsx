/* eslint-disable */
import React from 'react';
import Component from './Form';

/**
 * A basic tile device often used to group content or implement hierarchy.
 * 
 * ** Sub Components **
 *
 * @composedWith UiWrapper
*/
function Form(props) {
  return <Component {...props} />;
}

export default Form;
