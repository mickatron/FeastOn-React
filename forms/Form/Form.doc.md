

### Default Form.
```js
<Form>

</Form>
```