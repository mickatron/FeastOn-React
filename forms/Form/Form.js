import uiWrapper from '../../HOC/uiWrapper';

export default uiWrapper({
  name: 'Form',
  element: 'form',
});
