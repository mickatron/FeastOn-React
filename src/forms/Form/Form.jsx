import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import classNameType from 'js/shared/proptypes/className';

export default function Form({ className, children, disabled, valid }) {
  return (    
    <form className={ classnames('c-Form', className) }>
      { children }
    </form>
  );
}

Form.PropTypes = {
  className: classNameType,
  children: PropTypes.node.isRequired,
};