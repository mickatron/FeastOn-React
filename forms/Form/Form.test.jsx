import { shallow } from 'enzyme';
import React from 'react';
import Form from './Form';

describe('Form', () => {
  let component = null;

  it('renders', () => {
    component = shallow((
      <Form
        text="Form text"
        input={({ uid }) => (
          <input id={uid} />
        )}
      />
    ));
    expect(component).toMatchSnapshot();
  });
});
