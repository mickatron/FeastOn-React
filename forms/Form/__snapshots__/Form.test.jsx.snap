// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UiWrapper
    as="form"
    className={undefined}
    input={[Function]}
    modifier={undefined}
    text="Form text"
    widthClass={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Form",
      "input": [Function],
      "text": "Form text",
    },
    "ref": null,
    "rendered": null,
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Form",
        "input": [Function],
        "text": "Form text",
      },
      "ref": null,
      "rendered": null,
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
